### PR TITLE
Feat: Allow devServer.watchConfig to be configured

### LINKED ISSUE #515: Support Vagrant with VirtualBox
Hi,

I just tried to make styleguidist work with Vagrant, and it mostly works. There is a pretty deal-breaking issue with hot reloading, however.

When using the VirtualBox provider (which most people do because it's cross-platform) in Vagrant, you need to set `devServer: { poll: true }` for it to work because file system events do not work properly. See https://webpack.js.org/guides/development-vagrant/. This is not possible to set in styleguidist because of the way the server config is merged. See [create-server.js](https://github.com/styleguidist/react-styleguidist/blob/master/scripts/create-server.js#L15-L17).

I tried patching the file to include `poll: true`, and then it works just fine. It seems trivial to support this usecase by changing the config merging to the following:

```js
const webpackDevServerConfig = Object.assign({}, webpackConfig.devServer, {
		noInfo: true,
		compress: true,
		clientLogLevel: 'none',
		hot: true,
		quiet: true,
		watchOptions: Object.assign({}, {
			ignored: /node_modules/,
		}, webpackConfig.devServer ? webpackConfig.devServer.watchOptions : {}),
		contentBase: config.assetsDir,
		stats: webpackConfig.stats || {},
});
```

Shall I make a PR for this, or do you propose another solution?

### PR DESCRIPTION
**Pull request by esphen**
## Why

Currently, Vagrant with the VirtualBox provider does not work with styleguidist, as there are issues with webpack's default behavior. The way around this is to make webpack use a polling scheme for checking for changed files. This is currently not configurable in styleguidist.

## How

This PR makes the `devServer.watchConfig` configurable so that it is possible to use styleguidist with Vagrant

Fixes #515

### CODE REVIEW

File: scripts/create-server.js [hunk 1]
```diff
@@ -2,22 +2,29 @@
...
+				ignored: /node_modules/,
+			},
+			contentBase: config.assetsDir,
+			stats: webpackConfig.stats || {},
 		},
-		contentBase: config.assetsDir,
-		stats: webpackConfig.stats || {},
-	});
+		{
+			watchOptions: webpackConfig.devServer ? webpackConfig.devServer.watchOptions : {},
```
Comment (sapegin) [inline scripts/create-server.js]: I think this part isn’t required because webpack-merge should do recursive merge.

File: docs/Cookbook.md [hunk 2]
```diff
@@ -390,6 +390,22 @@ initialState = {
 </div>
 ```
 
+## How to use Vagrant with styleguidist?
```
Comment (sapegin) [inline docs/Cookbook.md]: styleguidist → Styleguidist

File: docs/Cookbook.md [hunk 3]
```diff
@@ -390,6 +390,22 @@ initialState = {
...
+
+First of all, make sure you read [this guide](https://webpack.js.org/guides/development-vagrant/) from the webpack documentation.
+
+The article mentions that you need to use `--watch-poll` when using the
```
Comment (sapegin) [inline docs/Cookbook.md]: I’d write it shorter like this: 

> Then enable polling in your webpack config:

**Comment by sapegin** [comment]
And thanks for the pull request! 🍕

File: scripts/create-server.js (hunk 1 above)
Comment (esphen) [inline scripts/create-server.js]: I'm sorry, not quite sure what you mean here. Would you like to merge the entire `webpackConfig.devServer` into the styleguidist dev server config, not just `watchConfig`? Here I am checking for `devServer` being defined in the user config so it doesn't throw when it isn't.

I have not used `webpack-merge` before, so I am assuming it works like `Object.assign` in most cases

Comment (sapegin) [inline scripts/create-server.js]: I’d try to remove this whole second block and see if it still works because webpack-merge does recursive merge and it should work ;-)

Comment (esphen) [inline scripts/create-server.js]: Ah, I see! Thanks for the ELI5. It does seems to work without the last block because of the recursive merging.

**Comment by esphen** [comment]
Amended the commit with your requested changes
